index.js

```diff
@@ -244,29 +244,51 @@ const create = (win, options) => {
 			*/
 			menu.popup(electron.remote ? electron.remote.getCurrentWindow() : win);
 		}
-	});
+	};
+
+	webContents(win).on('context-menu', onContextMenu);
+	return () => webContents(win).removeListener('context-menu', onContextMenu);
 };
 
 module.exports = (options = {}) => {
+	let isDisposed = false;
+	const disposables = [];
+	const createMenu = win => {
+		if (isDisposed) {
+			return;
+		}
+
+		const disposeMenu = create(win, options);
+		disposables.push(disposeMenu);
+	};
+
+	const dispose = () => {
+		disposables.forEach(dispose => dispose());
+		isDisposed = true;
```

Comment:
Instead of a boolean, just empty the array and check the length. You should empty the array regardless as you don't want to hold onto those forever.

Reply:
Hmm. I believe it would cause a chicken/egg problem and we'd never create the menu in the first place. Looking into it.

Reply:
Indeed, it's not that easy. 
Avoiding the flag requires either a completely different approach, or several more changes that seem unnecessarily complex.
It's really about bailing out in certain deferred edge cases, and the significant piece of information of the flag is: "No need to create the menu, the callsite already disposed it".

I already tried a couple variants, and I still believe the flag is the best solution. 

Should let go of the references despite of that.

Reply:
array length reset